docs/src/reference/architecture.md

```diff
@@ -0,0 +1,72 @@
+# Architecture
+
+A system architecture document is the starting point for many interested participants in a project, whether you intend contirbuting or simply want to understand how the software is structured. This documentation lays out the current design of Canonical Kubernetes, following the [C4 model]. 
+
+##  System context 
+
+This overview of Canononical Kubernetes demonstrates the interactions of Kubernetes with users and with other systems.
+
+```{kroki} ../assets/overview.puml
+```
+
+Two actors interact with the kubernetes snap:
+
+- **K8s admin**: The administrator of the cluster interacts directly with the Kubernetes API server. Out of the box our K8s distribution offers admin access to the cluster. That initial user is able to configure the cluster to match their needs and of course create other users that may or may not have the same privileges. The K8s admin is also able to deploy workloads running in the cluster.
+
+- **K8s user**: A user consuming the services hosted in the cluster. Users do not have access to the Kubernetes API server. They need to access the cluster through the options (nodeport, ingress, load-balancer) offered by the administrator who deployed the workload they are interested in.
+
+There are non-human users of the k8s snap. In this case that is the [K8s charm]. The K8s charm needs to drive the Kubernetes cluster and to orchestrate the multi-node clustering operations.
+
+A set of external systems need to be easily integrated with our k8s distribution. We have identified the following:
```

Comment:
Further up, k8s distribution is capitalized. I don't have a strong opinion here but we should be consistent with `k8s` or `K8s`.